vendor/github.com/algorand/websocket/conn.go

```diff
@@ -1148,8 +1138,8 @@ func (r *messageReader) Close() error {
 // NEVER USE THIS. An untrusted peer could send an explosively compressed message. A megabyte of zeros can compress down to about 1kB. Always use NextReader() and kill the connection if you read too long a message.
 // ReadMessage() will error out immediately unless Conn.Unsafe is true.
 func (c *Conn) ReadMessage() (messageType int, p []byte, err error) {
-	if !c.Unsafe {
-		return noFrame, nil, errors.New("websocket: ReadMessage is not safe but Conn.Unsafe is not set")
+	if c.readLimit == 0 && !c.Unsafe {
```

Comment:
Shouldn't this be OR instead of AND?

Reply:
if (c.Unsafe or c.readLimit != 0) then we let them do it.
if they haven't set readLimit and they haven't set Unsafe mode, they will be prevented from doing a read that could cause unbounded memory allocation.